Bind mount `/tmp/.X11-unix` for newer version of WSL2

Locietta opened the pull request:
Fix #56.

After [WSL 0.60.0 release](https://github.com/microsoft/WSL/releases/tag/0.60.0), microsoft switched to bind mount the `/tmp/.X11-unix` rather than symlink it as they previously do. This breaks wslg in Distrod for latest WSL on MS Store.

Distrod should bind mount for WSL 0.60.0, and keep the symlink strategy for older versions of WSL.

I've tested the fix on WSL 0.60.0, WSL 0.58.3 and in-box WSL, all of them worked ok.

**EDIT:** Before this PR is merged, you may use opt tar packaged by me for a while 👉 [opt_distrod.tar.gz](https://github.com/nullpo-head/wsl-distrod/files/8985168/opt_distrod.tar.gz)

herrwinfried commented:
are you sure you fixed it? it didn't work for me

Locietta commented:
Hi, @herrwinfried, I'm now using it for myself (WSL0.60.0 and archwsl) and doesn't see any issue🤔

Could you please test [opt_distrod.tar.gz](https://github.com/nullpo-head/wsl-distrod/files/8954538/opt_distrod.tar.gz) to see if it works? 
If it also fails, please paste the output of `ls /run/systemd/system/systemd-tmpfiles-setup.service.d` and `ls /tmp/.X11-unix` to see what's going wrong...

herrwinfried commented:
> Hi, @herrwinfried, I'm now using it for myself (WSL0.60.0 and archwsl) and doesn't see any issue🤔
> 
> Could you please test [opt_distrod.tar.gz](https://github.com/nullpo-head/wsl-distrod/files/8954538/opt_distrod.tar.gz) to see if it works? If it also fails, please paste the output of `ls /run/systemd/system/systemd-tmpfiles-setup.service.d` and `ls /tmp/.X11-unix` to see what's going wrong...

I'm using openSUSE TW but tmp/.x11-unix looks empty.

herrwinfried commented:
> Hi, @herrwinfried, I'm now using it for myself (WSL0.60.0 and archwsl) and doesn't see any issue🤔
> 
> Could you please test [opt_distrod.tar.gz](https://github.com/nullpo-head/wsl-distrod/files/8954538/opt_distrod.tar.gz) to see if it works? If it also fails, please paste the output of `ls /run/systemd/system/systemd-tmpfiles-setup.service.d` and `ls /tmp/.X11-unix` to see what's going wrong...

yes this works on the openSUSE TW windows store download you mentioned, thanks.
![image](https://user-images.githubusercontent.com/52379312/174963790-b2a20ca1-41a3-4e61-90d0-4f3b2e91b520.png)

Locietta commented:
I just notice a bug: when `systemd-tmpfiles --create` is executed, `/tmp/.X11-unix` will be cleaned up.

I've fixed it and updated the opt tar pinned at the beginning.

leoleoasd commented:
I tried the tarball and it gives the following error everytime i open wsl:
```

leoleoasd commented:
I'm using latest archlinux.

herrwinfried commented:
I downloaded the latest edited and it works for me. Could it be a missing step for arch linux. @leoleoasd 
![image](https://user-images.githubusercontent.com/52379312/175829735-8d538e95-457c-4fd5-bfc8-ed57fbc0fd33.png)

Locietta commented:
> I tried the tarball and it gives the following error everytime i open wsl:
> 
> ```
> C:\Users\Leo>wsl
> -zsh: symbol lookup error: /opt/distrod/lib/libpthread.so.0: undefined symbol: __libc_pthread_init, version GLIBC_PRIVATE
> ```
> 
> Maybe it's something with your compile environment?

It looks like you directly extract the tarball to `/opt`, which might cause the trouble. You are mixing the libs from repo owner (probably package his tarball on ubuntu) and me (on Arch).

Remove the original `/opt/distrod` and re-install the tarball should fix this. @leoleoasd 

And in case you can't open wsl shell, use `wsl -u root -e /opt/distrod/bin/distrod disable` to disable distrod from outside.

leoleoasd commented:
That worked for me. Thanks!

Locietta commented:
Seems like CI fails because no `/mnt/wslg/.X11-unix` on test runner for bind mount... while we don't have permission to create it on github runner😥 (see https://github.com/Locietta/wsl-distrod/runs/7183455218?check_suite_focus=true). 

It looks ok when I ran tests on my WSL2 though. Having no idea about how to solve the test failure.

nullpo-head commented:
Hi @Locietta ! I truly appreciate your contribution! And I'm sorry for my late response. 🙇‍♂️
Overall it looks good to me, but let me check it around tomorrow. I may ask you for some refactoring, or I could do it by myself after merging your contribution. Thanks so much again!

nullpo-head reviewed (approved):
Looks great to me. Thanks so much! I've add a little requests for improvements. I'd appreciate it if you could update it if you can.

Locietta commented on distrod/distrod/tests/integration_test.rs:
```diff
@@ -326,17 +326,17 @@ fn test_wslg_socket_is_available() {
+    test_old.args(&["exec", "--", "test", "-e", "/run/tmpfiles.d/x11.conf"]);
+    let result_old = test_old.status().unwrap();
+    
+    // mount systemd-tmpfiles-setup.service.d to bind-mount
```
done

dyharlan commented:
I tried installing the latest release, and it doesn't seem to fix the issue?

Locietta commented:
@dyharlan v0.1.7 looks good to me, could you check `ls /run/systemd/system/systemd-tmpfiles-setup.service.d` and `ls /tmp/.X11-unix`?

leoleoasd commented:
v0.1.7 gives me 
```
<3>init: (12) ERROR: CreateProcessEntryCommon:528: execvpe /opt/distrod/alias/usr/bin/zsh failed 2
<3>init: (12) ERROR: CreateProcessEntryCommon:538: Create process not expected to return
```

Is there something with my installation? I just delete all files inside opt/distrod and replace it with [opt_distrod.tar.gz](https://github.com/nullpo-head/wsl-distrod/releases/download/v0.1.7/opt_distrod.tar.gz)

gifnksm commented:
In my environment, `/run/tmpfiles.d/x11.conf` was still mounted after upgrading Distrod to v0.17 (even after disabling and enabling Distrod).
I removed everything under `/opt/distrod` and reinstalled Distrod again, and the problem no longer occurs.
